### PR TITLE
#7 PATCH /api/articles/:article_id completed

### PR DESCRIPTION
**Pull request by kam-cheng**
completed patch votes functionality

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -113,3 +113,60 @@ describe("GET/api/articles/:article_id", () => {
     expect(msg).toBe("Invalid input of article_id");
   });
 });
+describe("PATCH/api/articles/:article_id", () => {
+  test("202 - returns 202 status", async () => {
+    const votes = { inc_votes: 1 };
+    await request(app).patch("/api/articles/1").send(votes).expect(202);
```
Comment (douglashellowell) [inline __tests__/app.test.js:119]: `202` isn't the most appropriate status code here 
it usually indicates that the request has been accepted but that the outcome hasn't happened yet
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202

`201` would be more appropriate here

File: models/app.models.js [hunk 2]
```diff
@@ -35,3 +35,10 @@ exports.fetchUsers = async () => {
   const users = await db.query("SELECT username FROM users");
   return users.rows;
 };
+
+exports.updateVotes = async (articleId, votes) => {
+  const updatedVotes = await db.query(`UPDATE articles 
+  SET votes = votes + ${votes} 
+  WHERE article_id = ${articleId} RETURNING *;`);
```
Comment (douglashellowell) [inline models/app.models.js:42]: ⚠️⚠️careful!⚠️⚠️

you're exposing yourself to SQL injection here

what if i made this request

```json
{
   "inc_votes": "1; DROP DATABASE nc_news;"
}
```

😱

**Review by douglashellowell — CHANGES_REQUESTED** [review]
need to avoid SQL injection attacks in your models!

File: __tests__/app.test.js (hunk 1 above)
Comment (kam-cheng) [reply __tests__/app.test.js]: thanks douglas, I have changed this now :)

File: models/app.models.js (hunk 2 above)
Comment (kam-cheng) [reply models/app.models.js]: Thanks douglas! I can't believe I forgot this! updated now :)

File: controllers/app.controllers.js [hunk 3]
```diff
@@ -34,3 +35,15 @@ exports.getUsers = async (req, res) => {
...
 };
+
+exports.patchVotes = async (req, res, next) => {
+  const { inc_votes: votes } = req.body;
+  const { article_id: articleId } = req.params;
+  try {
+    const article = await updateVotes(articleId, votes);
+    res.status(201).send({ article });
+  } catch (err) {
+    console.log(err);
```
Comment (Angaca) [inline controllers/app.controllers.js:46]: No console.log in production code

**Review by Angaca — APPROVED** [review]
Good to go!

File: controllers/app.controllers.js (hunk 3 above)
Comment (kam-cheng) [reply controllers/app.controllers.js]: thanks Angaca - have now removed this :)

**Pull request merged**